packages/devtools-move/tasks/evm/wire/transactionExecutor.ts

```diff
@@ -103,12 +103,25 @@ export async function executeTransactions(
                 const provider: providers.JsonRpcProvider = new providers.JsonRpcProvider(rpcUrlsMap[eid])
                 const signer = accountEidMap[eid].signer
 
-                populatedTx.gasLimit = await provider.estimateGas(populatedTx)
+                // Try to estimate gas limit, if it fails, use a default value
+                const gasLimit = await provider
+                    .estimateGas(populatedTx)
+                    .then((gasLimit) => {
+                        return gasLimit
+                    })
+                    .catch((error) => {
+                        console.error(
+                            `Error estimating gas for transaction on chain ${network.chainName}-${network.env}: ${error}`
+                        )
+                        return BigNumber.from(1_000_000)
+                    })
+
+                populatedTx.gasLimit = gasLimit
                 populatedTx.gasPrice = accountEidMap[eid].gasPrice
                 populatedTx.nonce = accountEidMap[eid].nonce++
 
                 let sendTx: Promise<providers.TransactionResponse> | undefined = undefined
-                if (simulation === 'broadcast') {
+                if (simulation === 'broadcast' || simulation === 'dry-run') {
```

Comment:
nit: probably better to do `if (simulation !== 'calldata')` if we are are to have other modes later where we would have to keep adding to this list of conditionals.